core/src/App.html

```diff
@@ -304,22 +301,34 @@
           }
         });
         window.addEventListener('message', async e => {
+          const allIframes = [...Iframe.getAllIframes(node, this.get().modalIframe), {contentWindow: window, luigi: {viewUrl: window.location.href, trustedDomain: window.location.origin}}];
+          const isOriginWindow = (iframe) => iframe && (e.source === iframe.contentWindow);
+          const iframe = allIframes.find(isOriginWindow);
+          if (!(iframe && iframe.luigi && iframe.luigi.viewUrl)) return;
+
+          if ('luigi.navigate.ok' === e.data.msg) {
+            if (iframe.luigi.nextViewUrl && iFrameHelpers.urlMatchesTheDomain(iframe.luigi.nextViewUrl, e.origin)) {
+              iframe.luigi.viewUrl = iframe.luigi.nextViewUrl;
+              iframe.luigi.nextViewUrl = '';
+            }
+            config.navigateOk = true;
+          }
+
+          if (!iFrameHelpers.iframeIsSameDomain(iframe, e.origin)) return;
+
           if ('luigi.get-context' === e.data.msg) {
-            if (
-              this.get().modalIframe &&
-              e.source === this.get().modalIframe.contentWindow
-            ) {
+            if (isOriginWindow(this.get().modalIframe)) {
```

Comment:
I think we can use the same `if` statement below around line 377, which currently is:
```
         if (
              this.get().modalIframe &&
              e.source === this.get().modalIframe.contentWindow
            ) {
```